test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt

```diff
@@ -689,22 +942,109 @@ class QueryTests {
+
+    @Test
+    fun max_find_allNullValues() {
+        val assertions = { propertyDescriptor: PropertyDescriptor ->
+            realm.writeBlocking {
+                // Queries inside a write transaction produce live results which means they can be
+                // reused within the closure
+                val maxQuery = query(QuerySample::class)
+                    .max(propertyDescriptor.property.name, propertyDescriptor.clazz)
+
+                assertNull(maxQuery.find())
+                copyToRealm(getInstance(propertyDescriptor, QuerySample()))
+                copyToRealm(getInstance(propertyDescriptor, QuerySample()))
+                assertNull(maxQuery.find())
+            }
+
+            realm.query(QuerySample::class)
+                .max(propertyDescriptor.property.name, propertyDescriptor.clazz)
+                .find { maxValue -> assertNull(maxValue) }
+
+            // Make sure to delete all objects after assertions as aggregators to clean state and
+            // avoid "null vs 0" results when testing
+            cleanUpBetweenProperties()
+        }
+
+        // Iterate only over nullable properties and insert only null values in said properties
+        for (nullablePropertyDescriptor in nullablePropertyDescriptors) {
+            assertions(nullablePropertyDescriptor)
+        }
+    }
+
+    @Test
+    fun max_find() {
+        val assertions = { propertyDescriptor: PropertyDescriptor ->
+            val expectedMax =
+                expectedMax(propertyDescriptor.clazz, propertyDescriptor.property.returnType)
+
+            realm.writeBlocking {
+                // Queries inside a write transaction produce live results which means they can be
+                // reused within the closure
+                val maxQuery = query(QuerySample::class)
+                    .max(propertyDescriptor.property.name, propertyDescriptor.clazz)
+
+                assertNull(maxQuery.find())
+                copyToRealm(getInstance(propertyDescriptor, QuerySample(), 0))
+                copyToRealm(getInstance(propertyDescriptor, QuerySample(), 1))
+                assertEquals(expectedMax, maxQuery.find())
+            }
+
+            realm.query(QuerySample::class)
+                .max(propertyDescriptor.property.name, propertyDescriptor.clazz)
+                .find { maxValue -> assertEquals(expectedMax, maxValue) }
+
+            // Make sure to delete all objects after assertions as aggregators to clean state and
+            // avoid "null vs 0" results when testing
+            cleanUpBetweenProperties()
+        }
+
+        // Iterate over non-nullable properties only
+        for (propertyDescriptor in propertyDescriptors) {
+            assertions(propertyDescriptor)
+        }
+
+        // Iterate over nullable properties containing both null and non-null values
+        for (nullablePropertyDescriptor in nullablePropertyDescriptors) {
+            assertions(nullablePropertyDescriptor)
+        }
+    }
```

Comment:
Seems to be missing tests for unsupported field types like objects and lists

Reply:
Are these tests added somewhere now? I didn't find them ... unless it is some of the still pending TODOs. Is that for future a PR?

Reply:
Yes, all TODOs in this test file are to be added later when the corresponding functionality is added in upcoming PRs.